src/utils/actions.ts

```diff
@@ -0,0 +1,13 @@
+export type ActionWithPayload<T, P> = Action<T> & {
+  payload: P;
+};
+
+export type Action<Type> = {
+  type: Type;
+};
+
+export interface ActionCreator<T extends string, P> {
+  (...args: any[]): ActionWithPayload<T, P>;
+}
+
+// export const createAction = <Type extends string>(type: Type) => <Payload>(payload: Payload): ActionWithPayload<Type, Payload> => ({ type, payload });
```

Comment:
Remove commented code. Add this to an Issue if you want to look into creating an action generator (this snippet).